### PR TITLE
OpenAPI V2 add Content Type return

### PR DESCRIPTION
**Pull request by Jefftree**
Partially addresses https://github.com/kubernetes/kubernetes/issues/113903

OpenAPI V3 fix will be in a separate PR.

/assign @apelisse

### CODE REVIEW

File: pkg/handler/handler_test.go [hunk 1]
```diff
@@ -84,6 +85,13 @@ func TestRegisterOpenAPIVersionedService(t *testing.T) {
 		if resp.StatusCode != tc.respStatus {
 			t.Errorf("Accept: %v: Unexpected response status code, want: %v, got: %v", tc.acceptHeader, tc.respStatus, resp.StatusCode)
 		}
+		if resp.StatusCode != 200 {
+			continue
+		}
+
+		if resp.Header.Get("Content-Type") != tc.expectedContentType {
+			t.Errorf("Content-Type: Unexpected content type, want: %v, got: %v", tc.expectedContentType, resp.Header.Get("Content-Type"))
```
Comment (apelisse) [inline pkg/handler/handler_test.go:93]: Minor nit:
```suggestion
		if got := resp.Header.Get("Content-Type"); got != tc.expectedContentType {
			t.Errorf("Content-Type: Unexpected content type, want: %v, got: %v", tc.expectedContentType, got)
```

**Review by apelisse — COMMENTED** [review]
/lgtm
/approve
/hold if you want to fix the nit

**Comment by Jefftree** [comment]
Thanks, updated.

File: pkg/handler/handler_test.go [hunk 2]
```diff
@@ -84,6 +85,13 @@ func TestRegisterOpenAPIVersionedService(t *testing.T) {
 		if resp.StatusCode != tc.respStatus {
 			t.Errorf("Accept: %v: Unexpected response status code, want: %v, got: %v", tc.acceptHeader, tc.respStatus, resp.StatusCode)
 		}
+		if resp.StatusCode != 200 {
```
Comment (apelisse) [inline pkg/handler/handler_test.go:88]: I'll just mention that while I'm not super excited about this, I think it's fine given the current state of the test.

File: pkg/handler/handler.go [hunk 3]
```diff
@@ -203,6 +203,9 @@ func (o *OpenAPIService) RegisterOpenAPIVersionedService(servePath string, handl
 							return
 						}
 					}
+					contentType := accepts.Type + "/" + accepts.SubType
+					w.Header().Set("Content-Type", contentType)
```
Comment (liggitt) [inline pkg/handler/handler.go:207]: when this got pulled into kubernetes, we're seeing this error:

> Error from server (InternalError): Internal error occurred: mime: unexpected content after media subtype

Comment (liggitt) [reply pkg/handler/handler.go:207]: failed in https://github.com/kubernetes/kubernetes/pull/114869

Comment (liggitt) [reply pkg/handler/handler.go:207]: not sure if this PR or https://github.com/kubernetes/kube-openapi/pull/334/files was the cause... both were in the diff pulled in in that PR

**Pull request merged**